lib/dir.js

```diff
@@ -54,7 +52,22 @@ class DirFetcher extends Fetcher {
           npm_package_json: resolve(this.resolved, 'package.json'),
         },
       })
-    })
+    }
+    if (process.env._PACOTE_FROM_GIT_ === 'yes' && mani.scripts?.prepack) {
```

Comment:
I'm confused why this secret env is needed.  These are generally a code smell.

Reply:
I chose that because it seemed to be the chosen strategy for configuring the underlying NPM subprocess with special behavior when the git fetcher is invoking the dir fetcher (see `_PACOTE_NO_PREPARE_`). There could be better ways, like checking the spec of the package being built to see if it's a got spec, though. Idk why I didn't think of that.